core/src/test/java/hudson/slaves/NodeListTest.java

```diff
@@ -24,124 +24,52 @@
-            throw new UnsupportedOperationException();
-        }
-
-        public String getNodeDescription() {
-            throw new UnsupportedOperationException();
-        }
-
-        public Launcher createLauncher(TaskListener listener) {
-            throw new UnsupportedOperationException();
-        }
-
-        public int getNumExecutors() {
-            throw new UnsupportedOperationException();
-        }
-
-        public Mode getMode() {
-            throw new UnsupportedOperationException();
-        }
-
-        public Computer createComputer() {
-            throw new UnsupportedOperationException();
-        }
-
-        public Set<LabelAtom> getAssignedLabels() {
-            throw new UnsupportedOperationException();
-        }
-
-        public String getLabelString() {
-            throw new UnsupportedOperationException();
-        }
-
-        public void setLabelString(String labelString) throws IOException {
-            throw new UnsupportedOperationException();
-        }
-
-        public FilePath getWorkspaceFor(TopLevelItem item) {
-            throw new UnsupportedOperationException();
-        }
-
-        public FilePath getRootPath() {
-            throw new UnsupportedOperationException();
-        }
-
-        @Override
-        public Callable<ClockDifference, IOException> getClockDifferenceCallable() {
-            throw new UnsupportedOperationException();
-        }
-
-        public NodeDescriptor getDescriptor() {
-            throw new UnsupportedOperationException();
-        }
-
-		@Override
-		public DescribableList<NodeProperty<?>, NodePropertyDescriptor> getNodeProperties() {
-            throw new UnsupportedOperationException();
-		}
-    }
-
-    static class EphemeralNode extends DummyNode implements hudson.slaves.EphemeralNode {
-        public Node asNode() {
-            return this;
-        }
-    }
-
     @Test
     public void serialization() throws Exception {
-        NodeList nl = new NodeList(new DummyNode(), new EphemeralNode());
+        // create a normal and an ephemeral class, which should not be serialized
+        Node dummyNode = mock(Node.class, withSettings().serializable().defaultAnswer(CALLS_REAL_METHODS));
+        when(dummyNode.getNodeName()).thenReturn("node1");
+        Node ephemeralNode = mock(Node.class, withSettings().extraInterfaces(EphemeralNode.class).defaultAnswer(CALLS_REAL_METHODS));
+        when(ephemeralNode.getNodeName()).thenReturn("node2");
+        NodeList nl = new NodeList(dummyNode, ephemeralNode);
 
         File tmp = File.createTempFile("test","test");
         try {
             XmlFile x = new XmlFile(Jenkins.XSTREAM, tmp);
             x.write(nl);
 
             String xml = FileUtils.readFileToString(tmp, Charset.defaultCharset());
-            System.out.println(xml);
-            assertEquals(6,xml.split("\n").length);
+            // check that at least some content
+            assertTrue(xml.split("\n").length > 6);
```

Comment:
We used to test that we had exactly 6, now at least 7. I'm not sure to understand.

Reply:
Because the mock xml is bigger than 6 lines and I wanted to just check that it is not smaller than 6 lines but not that precise because I want not to fix this line for every mockito update. It is actually around 21 lines now because it is a much bigger "implementation" as before.